BraveRewardsTests/LedgerDatabaseTests.mm

```diff
@@ -144,6 +147,638 @@ - (void)testNonZeroExcludedCount
+  
+  const auto newerTips = [BATLedgerDatabase oneTimeTipsPublishersForMonth:month year:year];
+  XCTAssert([self countMustBeEqualTo:6 forEntityName:@"ContributionInfo"]);
+  XCTAssertEqual(newerTips.count, 3);
+}
+
+- (void)makeOneTimeTip:(NSString *)publisherId month:(const int)month year:(const int)year
+{
+  const auto probi = @"10000000000000000";
+  const BATRewardsCategory category = BATRewardsCategoryOneTimeTip;
+  const auto now = [[NSDate date] timeIntervalSince1970];
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertContributionInfo:probi month:month year:year date:now
+                                 publisherKey:publisherId category:category];
+  }];
+}
+
+- (void)makeOneTimeTip:(NSString *)publisherId month:(const int)month year:(const int)year
+                 probi:(NSString *)probi category:(BATRewardsCategory)category
+{
+  const auto now = [[NSDate date] timeIntervalSince1970];
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertContributionInfo:probi month:month year:year date:now
+                                 publisherKey:publisherId category:category];
+  }];
+}
+
+#pragma mark - Activity Info
+
+- (void)testInsertOrUpdateActivityInfoFromPublisherEmptyId
+{
+  const auto info = [self createBATPublisherInfo:@"111" reconcileStamp:111 percent:11 createActivityInfo:NO];
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+  
+  info.id = @"";
+  [self waitForFailedSaveAttempt:^{
+    [BATLedgerDatabase insertOrUpdateActivityInfoFromPublisher:info];
+  }];
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+}
+
+- (void)testInsertOrUpdateActivitiesInfoFromPublishers
+{
+  const auto p1 = [self createBATPublisherInfo:@"111" reconcileStamp:111 percent:11 createActivityInfo:NO];
+  const auto p2 = [self createBATPublisherInfo:@"222" reconcileStamp:222 percent:22 createActivityInfo:NO];
+  const auto p3 = [self createBATPublisherInfo:@"333" reconcileStamp:333 percent:33 createActivityInfo:NO];
+  
+  const auto publishers = @[p1, p2, p3];
+  
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertOrUpdateActivitiesInfoFromPublishers:publishers];
+  }];
+  
+  XCTAssert([self countMustBeEqualTo:3 forEntityName:@"ActivityInfo"]);
+}
+
+- (void)testDeleteActivityInfoWithPublisherID
+{
+  [self createBATPublisherInfo:@"1" reconcileStamp:10 percent:11 createActivityInfo:YES];
+  [self createBATPublisherInfo:@"2" reconcileStamp:10
+                                               percent:22 createActivityInfo:YES];
+  
+  XCTAssert([self countMustBeEqualTo:2 forEntityName:@"ActivityInfo"]);
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase deleteActivityInfoWithPublisherID:@"1" reconcileStamp:10];
+  }];
+  XCTAssert([self countMustBeEqualTo:1 forEntityName:@"ActivityInfo"]);
```

Comment:
Probably doesn't really matter, but do you think we should query the remaining activity info's to verify 2/10/22 is still there? All we verify by checking count is that we deleted 1 object, but didn't check which object was deleted.

Reply:
We can't access database object from tests, I haven't found solution to it yet